api.php

```diff
@@ -1,12 +1,57 @@
 <?php
 require_once('Books.php');
-if (preg_match("[^/api/books(|/)$]", $_SERVER['REQUEST_URI'])) {
-    Books::getInstance()->allBooks();
+function get($value, $default = null)
+{
+    return isset($value) ? $value : $default;
 }
-if (preg_match("[^/api/books/([0-9]{1,})(|/)$]", $_SERVER['REQUEST_URI'])) {
-    preg_match("/[0-9]{1,}/", $_SERVER['REQUEST_URI'], $matches);
-    Books::getInstance()->booksId($matches[0]);
+$user = get($_SERVER['PHP_AUTH_USER'], false);
+$pass = get($_SERVER['PHP_AUTH_PW'], false);
+if (!$user or !$pass) {
+    header('WWW-Authenticate: Basic realm');
 }
-if (preg_match("[^/api/books/count(|/)$]", $_SERVER['REQUEST_URI'])) {
-    Books::getInstance()->count();
+$login = Books::getInstance()->auth($user, $pass);
+if (!$login) {
+    header('WWW-Authenticate: Basic realm');
+    exit('incorrect login or password');
+}
+switch ($_SERVER['REQUEST_METHOD']) {
+    case "GET":
+        if (preg_match("[^/api/books(|/)$]", $_SERVER['REQUEST_URI'])) {
+            Books::getInstance()->allBooks();
+        }
+        if (preg_match("[^/api/books/([0-9]{1,})(|/)$]", $_SERVER['REQUEST_URI'])) {
+            preg_match("/[0-9]{1,}/", $_SERVER['REQUEST_URI'], $matches);
+            Books::getInstance()->booksId($matches[0]);
+        }
+        if (preg_match("[^/api/books/count(|/)$]", $_SERVER['REQUEST_URI'])) {
+            Books::getInstance()->count();
+        }
+        break;
+    case "POST":
+        if (preg_match("[^/api/books(|/)$]", $_SERVER['REQUEST_URI'])) {
+            $postdata = file_get_contents("php://input");
+            $array = json_decode($postdata, true);
+            if (is_array($array)) {

```

Comment:
Better to solve non array issue (one book input) with wrapped book with array.
Like:
```
$data = json_decode($postdata, true);
$data = is_array($data) ? $data else array($data);
```

Reply:
i think would be better to allow only array format of input data(for one or many books)

Reply:
During parsing input data i need origin keys. In case one book by input string format and use function 'array' or 'explode' i get next keys: 0 or 0,1,2.....